### PR TITLE
Clarify language around route services

### LINKED ISSUE #410: Route Service Binding phrasing
The Route Services section of [types of bindings](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#types-of-binding) of the spec specifies that:

`If a Platform supports route services, it MUST send a routable address, or endpoint, for the application along with the request to create a Service Binding using "bind_resource":{"route":"some-address.com"}.`

Taken as state, this implies that if the platform supports route services that it must send a routable address for all service bindings for any service. CF currently only sends the routable address if  `"required": ["route-forwarding"]` in the catalog for the service.

### PR DESCRIPTION
**Pull request by duglin**
See https://github.com/openservicebrokerapi/servicebroker/issues/410

Closes #410

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -1030,16 +1030,17 @@ response if the associated [Catalog](#catalog-management) entry for the
...
-There are a class of Service Offerings that intermediate requests to
-applications, performing functions such as rate limiting or authorization.
-
-If a Platform supports route services, it MUST send a routable address, or
-endpoint, for the application along with the request to create a Service
-Binding using `"bind_resource":{"route":"some-address.com"}`. A Service Broker
-MAY support configuration specific to an address using parameters; exposing
-this feature to users would require a Platform to support binding multiple
-routable addresses to the same Service Instance.
+Routing services are a class of Service Offerings that intermediate requests to
```
Comment (avade) [inline spec.md]: "Routing" or "Route" services?

Comment (duglin) [reply spec.md]: fixed! Got all spots.

File: spec.md [hunk 2]
```diff
@@ -1030,16 +1030,17 @@ response if the associated [Catalog](#catalog-management) entry for the
...
-this feature to users would require a Platform to support binding multiple
-routable addresses to the same Service Instance.
+Route services are a class of Service Offerings that intermediate requests to
+applications, performing functions such as rate limiting or authorization. To
+indicate support for route services, the catalog entry for the service MUST
+include the `"requires":["route_forwarding"]` property.
+
+When creating a route service type of Service Binding, a Platform MUST send
+a routable address, or endpoint, for the application along with the request to
+create a Service Binding using "bind_resource":{"route":"some-address.com"}.
```
Comment (angarg12) [inline spec.md]: Should this be between backticks `?

Comment (duglin) [reply spec.md]: yep - fixed!

**Comment by angarg12** [comment]
LGTM

File: spec.md [hunk 3]
```diff
@@ -1030,16 +1030,17 @@ response if the associated [Catalog](#catalog-management) entry for the
...
-
-If a Platform supports route services, it MUST send a routable address, or
-endpoint, for the application along with the request to create a Service
-Binding using `"bind_resource":{"route":"some-address.com"}`. A Service Broker
-MAY support configuration specific to an address using parameters; exposing
-this feature to users would require a Platform to support binding multiple
-routable addresses to the same Service Instance.
+Route services are a class of Service Offerings that intermediate requests to
+applications, performing functions such as rate limiting or authorization. To
+indicate support for route services, the catalog entry for the service MUST
```
Comment (mattmcneeney) [inline spec.md]: Capital S for Service

File: spec.md [hunk 4]
```diff
@@ -1030,16 +1030,17 @@ response if the associated [Catalog](#catalog-management) entry for the
...
+include the `"requires":["route_forwarding"]` property.
+
+When creating a route service type of Service Binding, a Platform MUST send
+a routable address, or endpoint, for the application along with the request to
+create a Service Binding using `"bind_resource":{"route":"some-address.com"}`.
+Service Broker MAY support configuration specific to an address using
```
Comment (mattmcneeney) [inline spec.md]: Broker**s**

File: spec.md [hunk 5]
```diff
@@ -1030,16 +1030,17 @@ response if the associated [Catalog](#catalog-management) entry for the
...
-There are a class of Service Offerings that intermediate requests to
-applications, performing functions such as rate limiting or authorization.
-
-If a Platform supports route services, it MUST send a routable address, or
-endpoint, for the application along with the request to create a Service
-Binding using `"bind_resource":{"route":"some-address.com"}`. A Service Broker
-MAY support configuration specific to an address using parameters; exposing
-this feature to users would require a Platform to support binding multiple
-routable addresses to the same Service Instance.
+Route services are a class of Service Offerings that intermediate requests to
```
[the remark this answers is not in the document]
Comment (duglin) [reply spec.md]: we do define it.  https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#terminology

File: spec.md (hunk 3 above)
Comment (duglin) [reply spec.md]: Fixed. We may need a new issues to catch all of these though

File: spec.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mattmcneeney) [reply spec.md]: Sorry, missed that one!

File: spec.md (hunk 4 above)
Comment (duglin) [reply spec.md]: fixed

**Comment by mattmcneeney** [comment]
LGTM

**Comment by fmui** [comment]
LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by zrob** [comment]
LGTM

**Pull request merged**